utils/utils.go

```diff
@@ -158,28 +163,55 @@ func GetRandomPortInRange(portRange string) uint32 {
 	}
 
 	mathrand.Seed(time.Now().UnixNano())
-	locHolder := mathrand.Intn(len(possible))
 
-	if len(possible[locHolder]) == 1 {
-		bindPort = uint32(possible[locHolder][0])
-	} else if len(possible[locHolder]) == 2 {
-		bindPort = uint32(mathrand.Intn(int(possible[locHolder][1]-possible[locHolder][0])) + int(possible[locHolder][0]))
-	}
+	if len(authPorts) > 0 {
+		bindPort = authPorts[mathrand.Intn(len(authPorts))]
+	} else {
+		locHolder := mathrand.Intn(len(possible))
 
+		if len(possible[locHolder]) == 1 {
+			bindPort = uint32(possible[locHolder][0])
+		} else if len(possible[locHolder]) == 2 {
+			bindPort = uint32(mathrand.Intn(int(possible[locHolder][1]-possible[locHolder][0])) + int(possible[locHolder][0]))
+		}
+
+	}
 	ln, err := net.Listen("tcp", fmt.Sprintf(":%d", bindPort))
 	if err != nil {
-		return GetRandomPortInRange(portRange)
+		if len(authPorts) == 1 {
+			return 0
```

Comment:
`0` will be a random port anyway. We want to still get a random port in a range (i.e. 1024-2048)

Reply:
No, my intention, in this case, is to block it. If all the ports in permitlisten are bind, you should not be able to create the tunnel

Reply:
bump

Reply:
We need to update the method to return an error then. I think returning 0 would still allow the tunnel to bind, wouldn't it?